### ISSUE TITLE
[rlgl] Support texture coordinates for triangles buffer

### ISSUE THREAD

**Issue by TheLumaio** [raysan5/raylib#556]
Binding textures using rlgl functions seems to only be available for `RL_QUADS`. Would be nice to have for `RL_TRIANGLES` in the future.

**Comment by raysan5** [comment]
Why do you need this feature? In those specific cases, is not better to just use your own `Mesh` data?

Actually, I was thinking about removing the `RL_TRIANGLES` internal buffer...

**Comment by TheLumaio** [comment]
I already implemented this in my own version of rlgl and haven't gotten a chance to write a pull request. The performance difference with mesh structs is insane. Meshes dropped framerates down to 15-25 fps while using the rlgl functions was as high as 700. The problem is Mesh (maybe it's Model) structs have their own VAO that needs to be bound every time you draw one creating insane performance loss.

**Comment by raysan5** [comment]
Wow! That's super weird! Actually, it should be the other way round! 

Mesh static data is preloaded in VRAM while using rlgl, data is created... and uploaded! in a per-frame basis! In both cases VAOs (mesh.vao/buffer vao) are enabled/disabled on drawing.

Do you have a test case for profiling?

**Comment by TheLumaio** [comment]
I don't have a test case. Maybe I didn't look into it enough but I still think it has to do with the single VAO versus the numerous VAOs from the 5k or so triangle meshes I had to generate to get the same result. The work flow for Mesh objects would have been
```
Bind VAO
Draw single triangle to VAO
Send data to GPU
```
for every single triangle mesh.
but manually using the triangles buffer would be more like
```
Bind VAO
Draw triangle to VAO
Draw triangle to VAO
Draw triangle to VAO
... and so on ...
Send data to GPU
```
Uploading all the triangles to a single mesh would cause them to be connected together strangely so that wasn't an option.

**Comment by raysan5** [comment]
Hey @TheLumaio, just reviewing all bugs and I found I didn't answer you here, sorry.

Actually both mechanisms are pretty similar. A public [`Mesh`](https://github.com/raysan5/raylib/blob/master/src/raylib.h#L502) struct is almost the same as a [`DynamicBuffer`](https://github.com/raysan5/raylib/blob/master/src/rlgl.h#L683) rlgl internal struct. Main difference is that `Mesh` is filled and managed by the user while `DynamicBuffer` is filled and managed internally by raylib.

`DynamicBuffer` is refilled per-frame while `Mesh` could be filled only once at beginning, consequently being more performant because you're not uploading data to the GPU every frame. In both cases, you always need to bound the respective VAO before drawing.

> Out of curiosity, what's your reasoning for removing the triangles buffer? Performance? or something else?

That's because currently rlgl manages internally 3 `DynamicBuffers` (LINES, TRIANGLES, QUADS), that results in a [number of issues](https://github.com/raysan5/raylib/wiki/raylib-internal-buffers-(2D-vs-3D)). Managing a single internal `DynamicBuffer` would solve some of those issues and simplify the code.

**Comment by raysan5** [comment]
Supported by commit https://github.com/raysan5/raylib/commit/97e40ced57489f4d66d4b9bc9fe213c388e1a827.
